### PR TITLE
Fix npc selling behavior if price is over INT_MAX.

### PR DESCRIPTION
**Pull request by Jedzkie**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Fix npc selling behavior if price is over INT_MAX.

**Issues addressed:** #2780

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/npc.c [hunk 1]
```diff
@@ -2805,19 +2805,18 @@ static int npc_selllist_sub(struct map_session_data *sd, struct itemlist *item_l
...
 
-	if( ( nd = npc->checknear(sd, map->id2bl(sd->npc_shopid)) ) == NULL ) {
+	if ((nd = npc->checknear(sd, map->id2bl(sd->npc_shopid))) == NULL)
 		return 1;
-	}
 
-	if( nd->subtype != SHOP ) {
-		if (!(nd->subtype == SCRIPT && nd->u.scr.shop && (nd->u.scr.shop->type == NST_ZENY || nd->u.scr.shop->type == NST_MARKET)))
+	if (nd->subtype != SHOP) {
+		if ((nd->subtype == SCRIPT && nd->u.scr.shop && (nd->u.scr.shop->type == NST_ZENY || nd->u.scr.shop->type == NST_MARKET)) == NULL)
```
Comment (4144) [inline src/map/npc.c:2819]: here error. you cant compare bool with NULL

Comment (Kenpachi2k13) [inline src/map/npc.c:2819]: Comparing the expression's result with `NULL` is incorrect. It has to be `true` or `false`. Or use keep using the "Not"-operator (`!`).

If you want to get rid of the "Not"-operator by all means, just invert the conditions:
`if (nd->subtype != SCRIPT || nd->u.scr.shop == NULL || (nd->u.scr.shop->type != NST_ZENY && nd->u.scr.shop->type != NST_MARKET))`

File: src/map/npc.c [hunk 2]
```diff
@@ -2826,73 +2825,65 @@ static int npc_selllist(struct map_session_data *sd, struct itemlist *item_list)
...
 			continue;
-		}
 
 		value = pc->modifysellvalue(sd, sd->inventory_data[idx]->value_sell);
 
 		z += (int64)value * entry->amount;
 	}
 
-	if( nd->master_nd ) { // Script-controlled shops
+	if (nd->master_nd) // Script-controlled shops
```
Comment (Kenpachi2k13) [inline src/map/npc.c:2854]:
```suggestion
	if (nd->master_nd != NULL) // Script-controlled shops.
```

File: src/map/npc.c [hunk 3]
```diff
@@ -2826,73 +2825,65 @@ static int npc_selllist(struct map_session_data *sd, struct itemlist *item_list)
...
 		nameid = sd->status.inventory[idx].nameid;
 
-		if (!nameid || !sd->inventory_data[idx] || sd->status.inventory[idx].amount < entry->amount) {
+		if (nameid == 0 || sd->inventory_data[idx] == 0 || sd->status.inventory[idx].amount < entry->amount)
 			return 1;
-		}
 
-		if (nd->master_nd) {
-			// Script-controlled shops decide by themselves, what can be sold and at what price.
+		if (nd->master_nd) // Script-controlled shops decide by themselves, what can be sold and at what price.
```
Comment (Kenpachi2k13) [inline src/map/npc.c:2846]:
```suggestion
		if (nd->master_nd != NULL) // Script-controlled shops decide by themselves, what can be sold and at what price.
```

File: src/map/npc.c [hunk 4]
```diff
@@ -2826,73 +2825,65 @@ static int npc_selllist(struct map_session_data *sd, struct itemlist *item_list)
...
+		if (duplicates[idx]) // Sanity check. The client sends each inventory index at most once [Haru]
 			return 1;
-		}
+
 		duplicates[idx] = true;
 
 		nameid = sd->status.inventory[idx].nameid;
 
-		if (!nameid || !sd->inventory_data[idx] || sd->status.inventory[idx].amount < entry->amount) {
+		if (nameid == 0 || sd->inventory_data[idx] == 0 || sd->status.inventory[idx].amount < entry->amount)
```
Comment (MishimaHaruna) [inline src/map/npc.c:2843]: Nitpick: The check for `sd->inventory_data[idx]` should be against NULL instead of 0 (since it's a pointer)

Comment (Kenpachi2k13) [reply src/map/npc.c]: Eagle eye...
*fixed*

Comment (MishimaHaruna) [reply src/map/npc.c]: Do you mean by skipping the verification entirely for script controlled shops? Verification of the item list to make sure that it isn't malformed/forged is still necessary for those shops, isn't it?

**Pull request merged**